expo-maps/ios/ExpoMaps/AppleMaps/AppleMapsPOI.swift

```diff
@@ -0,0 +1,101 @@
+import MapKit
+import ExpoModulesCore
+
+class AppleMapsPOI: NSObject, PointsOfInterests {
+    
+  private var pointsOfInterestSearchCompleter: AppleMapsPOISearchCompleter
+  private var pointsOfInterestSearchService: AppleMapsPOISearch
+  private var pointsOfInterestSearchController: AppleMapsPOISearchController
+  
+  private var mapView: MKMapView
+  private var markers: AppleMapsMarkers
+  
+  init(mapView: MKMapView, markers: AppleMapsMarkers) {
+    self.mapView = mapView
+    self.markers = markers
+    
+    pointsOfInterestSearchCompleter = AppleMapsPOISearchCompleter(delegate: nil)
+    pointsOfInterestSearchService = AppleMapsPOISearch(mapView: mapView, markers: markers)
+    pointsOfInterestSearchController = AppleMapsPOISearchController(searchService: pointsOfInterestSearchService)
+  }
+  
+  func fetchSearchCompletions(searchQueryFragment: String, promise: Promise) {
+    pointsOfInterestSearchCompleter.setSearchCompleterRegion(region: mapView.region)
+    pointsOfInterestSearchCompleter.autoComplete(searchQueryFragment: searchQueryFragment, promise: promise)
+  }
+  
+  func createSearchRequest(place: String) {
+    guard !place.isEmpty, let placeTitle = getPlaceTitleFromCompletion(place: place) else{
+      markers.detachAndDeleteMarkers()
```

Comment:
We have to create a separate array of markers inside `AppleMapsMarkers` because with current implementation all markers added by user will be removed when creatingSearchRequest is called

Reply:
Yes, obviously! Thank you :)